src/main/java/org/opentripplanner/inspector/vector/LayerParameters.java

```diff
@@ -0,0 +1,57 @@
+package org.opentripplanner.inspector.vector;
+
+import org.opentripplanner.api.mapping.PropertyMapper;
+
+/**
+ * Configuration options for a single vector tile layer.
+ */
+public interface LayerParameters<T extends Enum<T>> {
+  int MIN_ZOOM = 9;
+  int MAX_ZOOM = 20;
+  int CACHE_MAX_SECONDS = -1;
+  double EXPANSION_FACTOR = 0.25d;
+
+  /**
+   * User-visible name of the layer
+   */
+  String name();
+
+  /**
+   * Which {@link LayerBuilder} to use for fetching the geometries/objects for the layer.
+   */
+  T type();
+
+  /**
+   * Which {@link PropertyMapper} to use for mapping the object to the output properties.
+   */
+  String mapper();
```

Comment:
Is there a reason why this is a string and not an enum?

Reply:
It is dependent on the value of `type`. Each type has its own set of mappers.